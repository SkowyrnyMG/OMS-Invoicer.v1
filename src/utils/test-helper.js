import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
