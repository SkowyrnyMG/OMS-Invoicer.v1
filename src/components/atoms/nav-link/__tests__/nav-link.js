import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import NavLink from '../nav-link';

describe('NavLink', () => {
  test('should display login link in the document with correct styles', () => {
    snapShotTest(
      <NavLink path='/' linktype='login'>
        Login
      </NavLink>,
    );
  });
  test('should display logout link in the document with correct styles', () => {
    snapShotTest(
      <NavLink path='/' linktype='logout'>
        Login
      </NavLink>,
    );
  });
});
