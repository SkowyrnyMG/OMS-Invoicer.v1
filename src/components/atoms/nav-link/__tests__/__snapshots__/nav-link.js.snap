// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should display login link in the document with correct styles 1`] = `
.c0 {
  display: inline-block;
  padding: 1rem 2rem;
  margin-top: 0;
  font-weight: 700;
  color: #446DF6 !important;
  border-left: 1px solid #F0F0FD;
  background: none;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  border: none;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

<a
  class="c0"
  href="/"
  linktype="login"
>
  Login
</a>
`;

exports[`NavLink should display logout link in the document with correct styles 1`] = `
.c0 {
  display: inline-block;
  padding: 1rem 2rem;
  margin-top: 0;
  font-weight: 700;
  color: #DA2442 !important;
  border-left: 1px solid #F0F0FD;
  background: none;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  border: none;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

<a
  class="c0"
  href="/"
  linktype="logout"
>
  Login
</a>
`;
